### ISSUE TITLE
Unable to resolve module 'crypto' React 0.61.5

### ISSUE THREAD

**Issue by k00ksta** [auth0/react-native-auth0#276]
### Description
I have a fresh react-native application with nothing else in it but react-native-auth0 added. I have followed the documentation on their github and auth0 site to the letter. With my iOS emulator as soon as I import Auth0 into the class I receive an error stating:

Unable to resolve module 'crypto' from 'node_modules/crypto-js/core.js' : crypto could not be found within the project.

I have cleared watchman, node_modules and reset yarns cache with no success.

I do see crypto-js was added to the node modules when react-native-auth0 was installed.

### Screen Shots
<img width="241" alt="Screen Shot 2020-02-11 at 10 25 55 PM" src="https://user-images.githubusercontent.com/13239894/74303105-47f7c800-4d1e-11ea-9b90-9d4b879e5b5a.png">

<img width="296" alt="Screen Shot 2020-02-11 at 10 19 56 PM" src="https://user-images.githubusercontent.com/13239894/74303074-344c6180-4d1e-11ea-8bd7-660236c6a452.png">

### Reproduction
1) create a blank react native project using - react-native init some-project
2) yarn add react-native-auth0
3) pod install
4) add iOS requirements to AppDelegate.m from documentation
5) add iOS requirements to the plist.info file

6) run - react-native run-ios

### Environment

"react": "16.9.0",
"react-native": "0.61.5",
"react-native-auth0": "^2.2.0"

**Comment by f-roland** [comment]
Confirmed. this happens also on earlier versions of RN - I don't think crypto-js is usable in react-native. each time I wanted to use this, I had to use a polyfill

**Comment by cocojoe** [comment]
Thanks for reporting. @lbalmaceda can you pick this up, please.

**Comment by lbalmaceda** [comment]
This was tested locally with React Native `0.61.2` with and without the Hermes engine enabled and it worked fine on both iOS and Android platforms. I just went through that manual testing again using our quickstart sample, so I'm guessing the latest RN version is not supported right now, oh well... It's no surprise that every new RN release is breaking up things 😬. 

Let me migrate the sample app to the version you mentioned to corroborate this issue.

**Comment by lbalmaceda** [comment]
@k00ksta I couldn't repro it. I just tried all these with our [sample app](https://auth0.com/docs/quickstart/native/react-native/00-login), and they work fine.:

```
"dependencies": {
    "react": "16.8.1",
    "react-native": "0.61.2",
    "react-native-auth0": "^2.2.0"
  },
```
Run `yarn add react-native@0.61.5`

```
"dependencies": {
    "react": "16.8.1",
    "react-native": "0.61.5",
    "react-native-auth0": "^2.2.0"
  },
```
Run `yarn add react@16.9.0`
```
"dependencies": {
    "react": "16.9.0",
    "react-native": "0.61.5",
    "react-native-auth0": "^2.2.0"
  },
```

If you can provide more details, happy to look into it.

@f-roland On what version are you encountering that issue? Remember since `react-native-auth0` version 2, the [minimum required](https://github.com/auth0/react-native-auth0#compatibility-matrix) react-native SDK version is `0.60.5`.

**Comment by k00ksta** [comment]
I am not using your sample app. 

If you create a new react native project using "react-native init" and yarn add your current library it will fail.

"react": "16.9.0",
"react-native": "0.61.5",
"react-native-auth0": "^2.2.0"

I literally just downgraded my auth0 to 2.1.1 like the other user said and it works perfectly which does not include a node module for crypto-js but uses whatever you use previously.

also you should not only be changing react versions but deleting your node modules folder because you may still have crypto-js or some other dependancy included with that sample app.

I will pull the sample app now and see what is different from a stock react native project.

**Comment by f-roland** [comment]
@lbalmaceda well, up until 2.1.1, I can guarantee you it works with react-native 0.59.10 :)
You are free to drop support for pre-0.60 but as @k00ksta explained, this was also seen on more recent react-native versions.

did they include a crypto polyfill at some point ? I'm not aware of that but maybe I missed it.

the main issue is you are now relying on a lib that is primarily meant for browsers & node servers. so it has polyfills for the browsers, and uses node's crypto module on servers, but the way the code is written, it is bound to fail on react-native unless the haste resolver can somehow point to some other module to resolve `require("crypto")`

**Comment by cHaLkdusT** [comment]
Hi all, Using a bare RN project, I also encountered this issue. As a workaround, I tried using different release versions of `crypto-js` was able to make it work. I find myself resolving to lower version *v3.3.0*.

| | Version | Error |
| :--: | :--: | :-- |
| ❌ | v4.0.0 | error: Error: Unable to resolve module `crypto` from `node_modules/crypto-js/core.js`: crypto could not be found within the project. |
| ✅ | v3.3.0 |  |
| ❌ | v3.2.0 | Error: Native crypto module could not be used to get secure random number.

Summary of my `react-native info`:
```
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0
    react-native: 0.62.1 => 0.62.1
```
It seems like this isn't an RN issue but is more of the `crypto-js` dependency issue.

Hope this helps.

**Comment by lbalmaceda** [comment]
@cHaLkdusT According to their [changelog](https://github.com/brix/crypto-js/compare/3.2.1...develop#diff-04c6e90faac2675aa89e2176d2eec7d8R225) `3.3.0` is `3.1.9-1`. That would make sense since it's the version that we used when testing this. Although no change and no new release would be necessary for this library as [^3.1.9-1](https://github.com/auth0/react-native-auth0/blob/master/package.json#L51) would match 3.3.0 as well. 

![image](https://user-images.githubusercontent.com/3900123/80007227-5ab81700-849c-11ea-9e1c-53b3eb0083f5.png)


@k00ksta after installing the right crypto-js version (see above), is this still an issue for you?

**Comment by lbalmaceda** [comment]
Fixed to 3.3.0 on https://github.com/auth0/react-native-auth0/pull/299. Please, reopen if you encounter more issues after the next release.

**Comment by Uzef1997** [comment]
> ### Description
> I have a fresh react-native application with nothing else in it but react-native-auth0 added. I have followed the documentation on their github and auth0 site to the letter. With my iOS emulator as soon as I import Auth0 into the class I receive an error stating:
> 
> Unable to resolve module 'crypto' from 'node_modules/crypto-js/core.js' : crypto could not be found within the project.
> 
> I have cleared watchman, node_modules and reset yarns cache with no success.
> 
> I do see crypto-js was added to the node modules when react-native-auth0 was installed.
> 
> ### Screen Shots
> <img alt="Screen Shot 2020-02-11 at 10 25 55 PM" width="241" src="https://user-images.githubusercontent.com/13239894/74303105-47f7c800-4d1e-11ea-9b90-9d4b879e5b5a.png">
> 
> <img alt="Screen Shot 2020-02-11 at 10 19 56 PM" width="296" src="https://user-images.githubusercontent.com/13239894/74303074-344c6180-4d1e-11ea-8bd7-660236c6a452.png">
> 
> ### Reproduction
> 1. create a blank react native project using - react-native init some-project
> 2. yarn add react-native-auth0
> 3. pod install
> 4. add iOS requirements to AppDelegate.m from documentation
> 5. add iOS requirements to the plist.info file
> 6. run - react-native run-ios
> 
> ### Environment
> "react": "16.9.0",
> "react-native": "0.61.5",
> "react-native-auth0": "^2.2.0"

if you are still looking for a solution refer to this
https://stackoverflow.com/a/69265048/7017860
